homeassistant/components/whirlpool/climate.py

```diff
@@ -0,0 +1,238 @@
+"""Platform for climate integration."""
+import logging
+
+from whirlpool.aircon import Aircon, FanSpeed as AirconFanSpeed, Mode as AirconMode
+from whirlpool.auth import Auth
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    HVAC_MODE_COOL,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_SWING_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from .const import AUTH_INSTANCE_KEY, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up entry."""
+
+    auth: Auth = hass.data[DOMAIN][config_entry.entry_id][AUTH_INSTANCE_KEY]
+    said_list = auth.get_said_list()
```

Comment:
This looks like it is doing I/O so it should run in the executor.
```suggestion
    said_list = await hass.async_add_executor_job(auth.get_said_list)
```

Reply:
it just reads from memory. SAID data is fetched during auth.